library/src/main/java/io/ona/kujaku/activities/MapActivity.java

```diff
@@ -101,79 +102,95 @@ protected void onCreate(Bundle savedInstanceState) {
                 }
+            }
 
-                if (bundle.containsKey(Constants.PARCELABLE_KEY_CAMERA_TARGET_LATLNG)) {
-                    cameraTargetLatLng = bundle.getParcelable(Constants.PARCELABLE_KEY_CAMERA_TARGET_LATLNG);
-                }
+            if (bundle.containsKey(Constants.PARCELABLE_KEY_MAX_ZOOM)) {
+                maxZoom = bundle.getDouble(Constants.PARCELABLE_KEY_MAX_ZOOM);
+            }
 
-                if (bundle.containsKey(Constants.PARCELABLE_KEY_CAMERA_ZOOM)) {
-                    cameraZoom = bundle.getDouble(Constants.PARCELABLE_KEY_CAMERA_ZOOM);
-                }
+            if (bundle.containsKey(Constants.PARCELABLE_KEY_MIN_ZOOM)) {
+                minZoom = bundle.getDouble(Constants.PARCELABLE_KEY_MIN_ZOOM);
+            }
 
-                if (bundle.containsKey(Constants.PARCELABLE_KEY_CAMERA_TILT)) {
-                    cameraTilt = bundle.getDouble(Constants.PARCELABLE_KEY_CAMERA_TILT);
-                }
+            if (bundle.containsKey(Constants.PARCELABLE_KEY_CAMERA_TARGET_LATLNG)) {
+                cameraTargetLatLng = bundle.getParcelable(Constants.PARCELABLE_KEY_CAMERA_TARGET_LATLNG);
+            }
 
-                if (bundle.containsKey(Constants.PARCELABLE_KEY_CAMERA_BEARING)) {
-                    cameraBearing = bundle.getDouble(Constants.PARCELABLE_KEY_CAMERA_BEARING);
-                }
+            if (bundle.containsKey(Constants.PARCELABLE_KEY_CAMERA_ZOOM)) {
+                cameraZoom = bundle.getDouble(Constants.PARCELABLE_KEY_CAMERA_ZOOM);
+            }
 
-                if (stylesArray != null) {
-                    currentStylePath = stylesArray[0];
-                    if (currentStylePath != null && !currentStylePath.isEmpty()) {
-                        currentStylePath = new MapBoxStyleStorage()
-                                .getStyleURL(currentStylePath);
-                        mapboxStyleJSON = getStyleJSON(currentStylePath);
-
-                        // Extract kujaku meta-data
-                        try {
-                            sortFields = extractSortFields(mapboxStyleJSON);
-                            dataLayers = extractSourceNames(mapboxStyleJSON, sortFields);
-                            featuresMap = extractLayerData(mapboxStyleJSON, dataLayers);
-                            featuresMap = sortData(featuresMap, sortFields);
-                            displayInitialFeatures(featuresMap);
-                        } catch (JSONException e) {
-                            Log.e(TAG, Log.getStackTraceString(e));
-                        }
+            if (bundle.containsKey(Constants.PARCELABLE_KEY_CAMERA_TILT)) {
+                cameraTilt = bundle.getDouble(Constants.PARCELABLE_KEY_CAMERA_TILT);
+            }
+
+            if (bundle.containsKey(Constants.PARCELABLE_KEY_CAMERA_BEARING)) {
+                cameraBearing = bundle.getDouble(Constants.PARCELABLE_KEY_CAMERA_BEARING);
```

Comment:
What about https://www.mapbox.com/mapbox-gl-js/style-spec/#root-bearing

Reply:
I will look at this

Reply:
Add this also as an issue ❗️ ❗️